src/components/stimuli/Stimuli.tsx

```diff
@@ -29,15 +41,104 @@ const Stimuli: React.FC = () => {
       }
     }
 
-    fetchStimuliUrls();
-  }, []);
+    if (!isUploading) fetchStimuliUrls();
+  }, [isUploading, file]);
+
+  const fileChangedHandler = (event: any) => {
+    setFileInputText(event.target.files[0].name);
+    setFile(event.target.files[0]);
```

Comment:
These will need to be changed to use the combined `selectedFile` state as described in my other comment.